### PR TITLE
RHCLOUD-13446: Add Config Manager Check to remediations execute & preflight logic

### PR DESCRIPTION
**Pull request by aleccohan**
Added Config Manager check to remediations execute & pre-flight check logic.

JIRA: https://issues.redhat.com/browse/RHCLOUD-13502
   
    ## Secure Coding Practices Checklist GitHub Link
    - https://github.com/RedHatInsights/secure-coding-checklist

    ## Secure Coding Checklist
    - [ ] Input Validation
    - [ ] Output Encoding
    - [ ] Authentication and Password Management
    - [ ] Session Management
    - [ ] Access Control
    - [ ] Cryptographic Practices
    - [ ] Error Handling and Logging
    - [ ] Data Protection
    - [ ] Communication Security
    - [ ] System Configuration
    - [ ] Database Security
    - [ ] File Management
    - [ ] Memory Management
    - [ ] General Coding Practices

### CODE REVIEW

File: src/remediations/fifi.js [hunk 1]
```diff
@@ -37,6 +39,12 @@ exports.checkSmartManagement = async function (remediation, smart_management) {
         return false;
     }
 
+    const rhcStates = await configManager.getCurrentState();
```
Comment (jharting) [inline src/remediations/fifi.js:42]: As this is unrelated to smart management it would be a bit cleaner either to do the check outside of the function or rename the function to have a more general name. Also, since this is a API call there is a potential to optimize performance by making the call in parallel with other slow operations (API calls, db queries, ...).

**Review by Jason-RH — APPROVED** [review]

File: src/remediations/controller.fifi.js [hunk 2]
```diff
@@ -26,7 +26,18 @@ exports.connection_status = errors.async(async function (req, res) {
         throw new errors.Forbidden();
     }
 
-    const status = await fifi.getConnectionStatus(remediation, req.identity.account_number, req.entitlements.smart_management);
+    const rhcEnabled = await fifi.checkRhcEnabled();
+
+    if (!rhcEnabled && !req.entitlements.smart_management) {
```
Comment (jharting) [inline src/remediations/controller.fifi.js:31]: Is `&& !req.entitlements.smart_management` there because you can only enable remediations in RHC if you have smart management?

Comment (jharting) [reply src/remediations/controller.fifi.js:31]: Well either way I think the implication is that if you purchase smart management we may run stuff on your hosts using RHC regardless of whether you have that option enabled in the RHC dashboard.

Comment (aleccohan) [reply src/remediations/controller.fifi.js:31]: oh!  @jharting so the reason I added that to the check is to fail if rhc is not enabled AND the customer doesn't have the smart_management entitlement, I was under the impression that if you didn't have rhcEnabled on the dashboard and didn't have the entitlement you wouldn't be able to run any playbooks  (either through RHC or Satellite)

File: src/remediations/controller.fifi.js [hunk 3]
```diff
@@ -26,7 +26,18 @@ exports.connection_status = errors.async(async function (req, res) {
         throw new errors.Forbidden();
     }
 
-    const status = await fifi.getConnectionStatus(remediation, req.identity.account_number, req.entitlements.smart_management);
+    const rhcEnabled = await fifi.checkRhcEnabled();
```
Comment (jharting) [inline src/remediations/controller.fifi.js:29]: might be better to do this in parallel with the db query to speed up the operation

File: src/remediations/__snapshots__/fifi.integration.js.snap [hunk 4]
```diff
@@ -75,6 +75,73 @@ exports[`FiFi connection status get connection status with false smartManagement
...
+            \\"system_count\\": 1,
+            \\"connection_status\\": \\"no_executor\\"
+        },
+        {
+            \\"endpoint_id\\": null,
+            \\"executor_id\\": null,
+            \\"executor_type\\": \\"RHC\\",
+            \\"executor_name\\": null,
+            \\"system_count\\": 2,
+            \\"connection_status\\": \\"disabled\\"
```
Comment (jharting) [inline src/remediations/__snapshots__/fifi.integration.js.snap:139]: Even though this is reported as disabled to the user, if they submit the pre-flight will remediations still attempt to execute playbook on these hosts? If not, pls point me to the code that takes care about that.

File: src/remediations/controller.fifi.js (hunk 2 above)
Comment (jharting) [reply src/remediations/controller.fifi.js:31]: Have you considered using the new DISABLED state instead? I.e. regardless of whether I have smart management or not, if I have rhc-connected hosts with RHC disabled, I see x hosts in the rhc-connected-disabled bucket in preflight. I think that ways users are more likely to find the switch they need to flip than if all they get is a 403.

File: src/remediations/__snapshots__/fifi.integration.js.snap (hunk 4 above)
Comment (aleccohan) [reply src/remediations/__snapshots__/fifi.integration.js.snap:139]: Hey!  so yes.  the "execute" method will only run the executors marked with 'connected'

--> right here we filter out all NON connected executors https://github.com/RedHatInsights/insights-remediations/blob/master/src/remediations/fifi.js#L140

File: src/remediations/controller.fifi.js (hunk 3 above)
Comment (aleccohan) [reply src/remediations/controller.fifi.js:29]: Will do!  making the change now.

File: src/remediations/__snapshots__/fifi.integration.js.snap (hunk 4 above)
Comment (jharting) [reply src/remediations/__snapshots__/fifi.integration.js.snap:139]: I see, thanks!

File: src/remediations/controller.fifi.js (hunk 2 above)
Comment (aleccohan) [reply src/remediations/controller.fifi.js]: I agree!  I like this alot better than sending a 403.  The newly pushed changes should remedy this!

File: src/remediations/controller.fifi.js [hunk 5]
```diff
@@ -45,7 +53,18 @@ exports.executePlaybookRuns = errors.async(async function (req, res) {
...
+
+    if (!rhcEnabled && !req.entitlements.smartManagement) {
+        throw new errors.Forbidden();
+    }
```
Comment (jharting) [inline src/remediations/controller.fifi.js:60]: The same changes you made to `connection_status` are also applicable here.

Comment (aleccohan) [reply src/remediations/controller.fifi.js]: @jharting mirrored changes from connection_status in "execute" logic.

**Review by jharting — APPROVED** [review]
